xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCRoom.java

```diff
@@ -1402,23 +1402,31 @@ public void sendPublicMessage(Message message, MUCRole senderRole) throws Forbid
      * conversation between room occupants or IQ packets when an occupant wants to send IQ packets
      * to other room occupants.
      *
+     * If the system property xmpp.muc.allowpm.blockall is set to true, this will block all private packets
+     * However, if this property is false (by default) then it will allow non-message private packets.
+     *
      * @param packet The packet to send.
      * @param senderRole the role of the user that is trying to send a public message.
      * @throws NotFoundException If the user is sending a packet to a room JID that does not exist.
      * @throws ForbiddenException If a user of this role is not permitted to send private messages in this room.
      */
     public void sendPrivatePacket(Packet packet, MUCRole senderRole) throws NotFoundException, ForbiddenException {
-        switch (senderRole.getRole()) { // intended fall-through
-            case none:
-                throw new ForbiddenException();
-            default:
-            case visitor:
-                if (canSendPrivateMessage().equals( "participants" )) throw new ForbiddenException();
-            case participant:
-                if (canSendPrivateMessage().equals( "moderators" )) throw new ForbiddenException();
-            case moderator:
-                if (canSendPrivateMessage().equals( "none" )) throw new ForbiddenException();
+
+        if (packet instanceof Message || JiveProperties.getInstance().getBooleanProperty("xmpp.muc.allowpm.blockall", false)){
```

Comment:
`JiveProperties` shouldn't be used anymore (we aught to deprecate that, I think). Instead, try using a `SystemProperty`. That's the new implementation for properties. The i18n line that you added will be picked up by that (and won't be in the API that you're now using).